enterprise/javaee.wildfly/src/org/netbeans/modules/javaee/wildfly/ide/WildflyJ2eePlatformFactory.java

```diff
@@ -137,23 +156,26 @@ public J2eePlatformImplImpl(WildFlyProperties properties) {
         @Override
         public Set<org.netbeans.api.j2ee.core.Profile> getSupportedProfiles() {
             if (this.properties.isWildfly()) {
-                if(this.properties.isServletOnly()) {
+                if (this.properties.isServletOnly()) {
+                    if (this.properties.getServerVersion().compareToIgnoreUpdate(WildflyPluginUtils.WILDFLY_27_0_0) >= 0) {
+                        return Collections.unmodifiableSet(JAKARTAEE_WEB_PROFILES);
+                    }
                     return Collections.unmodifiableSet(WILDFLY_WEB_PROFILES);
                 }
+                if (this.properties.getServerVersion().compareToIgnoreUpdate(WildflyPluginUtils.WILDFLY_27_0_0) >= 0) {
+                    return Collections.unmodifiableSet(JAKARTAEE_FULL_PROFILES);
+                }
                 return Collections.unmodifiableSet(WILDFLY_PROFILES);
             }
-            if(this.properties.getServerVersion().compareToIgnoreUpdate(WildflyPluginUtils.EAP_7_0) >= 0) {
+            if (this.properties.getServerVersion().compareToIgnoreUpdate(WildflyPluginUtils.EAP_7_0) >= 0) {
                 return Collections.unmodifiableSet(WILDFLY_PROFILES);
             }
```

Comment:
on web projects and WF 27+ this returns only JAKARTAEE_FULL_PROFILES, which causes the combo box to remain empty since the UI in ServerSelectionHelper is filtering the FULL profiles out.

given that WILDFLY_PROFILES contains both web and full profiles, I believe how this is supposed to work is to return JAKARTAEE_FULL_PROFILES + JAKARTAEE_WEB_PROFILES in line 166

Reply:
something like:
```diff
diff --git a/enterprise/javaee.wildfly/src/org/netbeans/modules/javaee/wildfly/ide/WildflyJ2eePlatformFactory.java b/enterprise/javaee.wildfly/src/org/netbeans/modules/javaee/wildfly/ide/WildflyJ2eePlatformFactory.java
index c33683b..0613e47 100644
--- a/enterprise/javaee.wildfly/src/org/netbeans/modules/javaee/wildfly/ide/WildflyJ2eePlatformFactory.java
+++ b/enterprise/javaee.wildfly/src/org/netbeans/modules/javaee/wildfly/ide/WildflyJ2eePlatformFactory.java
@@ -163,7 +163,10 @@
                     return Collections.unmodifiableSet(WILDFLY_WEB_PROFILES);
                 }
                 if (this.properties.getServerVersion().compareToIgnoreUpdate(WildflyPluginUtils.WILDFLY_27_0_0) >= 0) {
-                    return Collections.unmodifiableSet(JAKARTAEE_FULL_PROFILES);
+                    Set<org.netbeans.api.j2ee.core.Profile> allJakarta = new HashSet<>();
+                    allJakarta.addAll(JAKARTAEE_FULL_PROFILES);
+                    allJakarta.addAll(JAKARTAEE_WEB_PROFILES);
+                    return Collections.unmodifiableSet(allJakarta);
                 }
                 return Collections.unmodifiableSet(WILDFLY_PROFILES);
             }

```

Reply:
Yes that looks right. The WEB profile is a subset of the FULL profile, so a server supporting the FULL profile will also support the WEB profile (at least if the developers are not completely nuts).

Reply:
if you want to cherry pick:
https://github.com/mbien/netbeans/commit/02e614160423225f61180ed4b13f89a29a0f10fb
I can could prepare a followup PR or you commit it yourself, whatever is more convenient for you.

https://github.com/mbien/netbeans/tree/pr-4692 is a copy of this PR + my commit

Reply:
or lets merge this once tests are sufficiently green. I prepare a followup.